Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPITabs.mm

```diff
@@ -1396,6 +1461,160 @@
+        @"browser.test.assertEq(response.content, pinkValue)",
+
+        @"await browser.tabs.removeCSS(tabId, { allFrames: false, file: 'styles.css' })",
+        @"response = await browser.test.assertSafeResolve(() => browser.tabs.sendMessage(tabId, { content: 'Get background color' }))",
+        @"browser.test.assertEq(response.content, whiteValue)",
+
+        @"await browser.tabs.insertCSS(tabId, { allFrames: true, file: 'styles.css' })",
+        @"response = await browser.test.assertSafeResolve(() => browser.tabs.sendMessage(tabId, { content: 'Get background color' }))",
+        @"browser.test.assertEq(response.content, pinkValue)",
+
+        @"await browser.tabs.removeCSS(tabId, { allFrames: true, file: 'styles.css' })",
+        @"response = await browser.test.assertSafeResolve(() => browser.tabs.sendMessage(tabId, { content: 'Get background color' }))",
+        @"browser.test.assertEq(response.content, whiteValue)",
+
+        @"await browser.tabs.insertCSS(tabId, { frameId: 0, file: 'styles.css' })",
+        @"response = await browser.test.assertSafeResolve(() => browser.tabs.sendMessage(tabId, { content: 'Get background color' }))",
+        @"browser.test.assertEq(response.content, pinkValue)",
+
+        @"await browser.tabs.removeCSS(tabId, { frameId: 0, file: 'styles.css' })",
+        @"response = await browser.test.assertSafeResolve(() => browser.tabs.sendMessage(tabId, { content: 'Get background color' }))",
+        @"browser.test.assertEq(response.content, whiteValue)",
+
+        @"await browser.tabs.insertCSS(tabId, { frameId: 0, code: 'body { background-color: blue !important }' })",
+        @"response = await browser.test.assertSafeResolve(() => browser.tabs.sendMessage(tabId, { content: 'Get background color' }))",
+        @"browser.test.assertEq(response.content, blueValue)",
+
+        @"await browser.tabs.removeCSS(tabId, { frameId: 0, code: 'body { background-color: blue !important }' })",
+        @"response = await browser.test.assertSafeResolve(() => browser.tabs.sendMessage(tabId, { content: 'Get background color' }))",
+        @"browser.test.assertEq(response.content, whiteValue)",
+
+        // Stylesheet being removed should match the one inserted.
+        @"await browser.tabs.insertCSS(tabId, { frameId: 0, code: 'body { background-color: blue !important }' })",
+        @"await browser.tabs.removeCSS(tabId, { allFrames: true, code: 'body { background-color: blue !important }' })",
+        @"response = await browser.test.assertSafeResolve(() => browser.tabs.sendMessage(tabId, { content: 'Get background color' }))",
+        @"browser.test.assertEq(response.content, blueValue)",
+
+        @"browser.test.notifyPass()"
+    ]);
+
+    auto extension = adoptNS([[_WKWebExtension alloc] _initWithManifestDictionary:manifest resources:@{ @"background.js": backgroundScript, @"content.js": tabsContentScriptManifestV2, @"styles.css": css }]);
```

Comment:
`@"content.js": tabsContentScriptManifestV2` is I assume the bug you found? This should be a script not the manifest.

Reply:
yea that was it